### ISSUE TITLE
[a11y]: DataTable [with-radio-selection] - keyboard nav radio button bug

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#14268]
### Package

@carbon/react

### Browser

FF

### Package version

@carbon/react@1.32.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

JAWS

### Description
Radio buttons in the table

Unable to navigate through the Radio Buttons using Up/Down arrow keys

When the focus is on the currently selected Radio button, We are unable to navigate to the other radio buttons using up / down arrow keys.

### Recommended Fix
This issue might be happening due to the name attribute being different for each radio button. Use the same name attribute for all the radio buttons.

When the standard keyboard navigation method is not supported, it impacts the user’s ability to interact with page controls. This affects screen reader users. 

### WCAG 2.1 Violation
2.1.1 Keyboard navigation

### Reproduction/example
Record `Number 74` in the below table:
https://airtable.com/shrFsaeHhImOkuVNR

https://react.carbondesignsystem.com/?path=/story/components-datatable-selection--with-radio-selection

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
